### PR TITLE
Background task component

### PR DESCRIPTION
**Pull request by gonzalocasas**
Added a new class `BackgroundWorker` and a new Grasshopper component `BackgroundTask` to execute long-running tasks:

background-task

The original inspiration for this feature is Speckle's awesome [`GrasshopperAsyncComponent`](https://github.com/specklesystems/GrasshopperAsyncComponent) (by @didimitrie @AlanRynne et al), but in Python form instead of C# (and also fitted into the COMPAS ecosystem).

NOTE: This PR is based off of `docs` branch, but if that is merged first, I'll rebase to `main`.

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [x] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_eve/ghpython/components/Ce_BackgroundTask/code.py [hunk 1]
```diff
@@ -5,57 +5,57 @@
```
Comment (chenkasirer) [inline src/compas_eve/ghpython/components/Ce_BackgroundTask/code.py:5]: `Ce` on purpose in file name?

**Review by chenkasirer — APPROVED** [review]
LGTM!

File: src/compas_eve/ghpython/components/Ce_BackgroundTask/code.py (hunk 1 above)
Comment (gonzalocasas) [reply src/compas_eve/ghpython/components/Ce_BackgroundTask/code.py:5]: Yep! It's `C(ompas)E(ve)`, I think we use this convention everywhere, don't we? https://github.com/compas-dev/compas_fab/tree/main/src/compas_fab/ghpython/components :P

Comment (chenkasirer) [reply src/compas_eve/ghpython/components/Ce_BackgroundTask/code.py:5]: damn I've been doing these uppercase all along. my bad..

Comment (gonzalocasas) [reply src/compas_eve/ghpython/components/Ce_BackgroundTask/code.py:5]: ahh... well, we never really wrote defined if it has to be upper or lower case :)

**Pull request merged**